### PR TITLE
Fix stuck coroutines on fini'd proxy object, improve test coverage

### PR DESCRIPTION
**Pull request by jnwatson**

### CODE REVIEW

File: synapse/tests/test_telepath.py [hunk 1]
```diff
@@ -174,9 +179,35 @@ def test_telepath_basics(self):
             prox = await s_telepath.openurl('tcp://127.0.0.1/foo', port=addr[1])
             genr = prox.corogenr(3)
             self.eq([0, 1, 2], [x async for x in await genr])
-            # To act the same as a local object, should be:
+            # To act the same as a local object, would be:
             # self.eq([0, 1, 2], [x async for x in genr])
 
+            aitr = (await prox.corogenr('fred')).__aiter__()
+            await self.asyncraises(s_exc.SynErr, aitr.__anext__())
```
Comment (vEpiphyte) [inline synapse/tests/test_telepath.py]: is there a specific error raised here?

File: synapse/tests/test_cortex.py [hunk 2]
```diff
@@ -1325,3 +1325,8 @@ def test_cortex_int_indx(self):
             self.len(0, core.eval('testint<10'))
             self.len(0, core.eval('testint<20'))
             self.len(1, core.eval('testint<30'))
+
+            list(core.eval('[testint=99999]'))
```
Comment (vEpiphyte) [inline synapse/tests/test_cortex.py]: what's this for?

File: synapse/tests/test_telepath.py (hunk 1 above)
Comment (jnwatson) [reply synapse/tests/test_telepath.py]: yes, yes there is.

File: synapse/tests/test_cortex.py (hunk 2 above)
Comment (jnwatson) [reply synapse/tests/test_cortex.py]: Imagine a different backing store might implement open vs. closed interval differently.

File: synapse/tests/test_cortex.py [hunk 3]
```diff
@@ -1381,3 +1381,8 @@ def test_cortex_int_indx(self):
             self.len(1, core.eval('teststr +teststr:tick<201808021202'))
             self.len(0, core.eval('teststr +teststr:tick<201808021201'))
             self.len(0, core.eval('teststr +teststr:tick<201808021200'))
+
+            list(core.eval('[testint=99999]'))
+            self.len(1, core.eval('testint<=20'))
+            self.len(1, core.eval('testint>20'))
```
Comment (vEpiphyte) [inline synapse/tests/test_cortex.py]: Do we want to add a `>=` test here as well, just  ensure we're well behaved at that boundary?

Comment (jnwatson) [reply synapse/tests/test_cortex.py]: ok
